src/router/utils/getHashString.es6.js

```diff
@@ -0,0 +1,9 @@
+'use strict';
+
+export default function (url) {
+	const hashMatch = url.match(/#([^?]*)/);
```

Comment:
Making the same comment here that I made to you offline (so that it's noted somewhere). The first regex in `getQueryString.es6.js` strips everything from (including) the '#' character to the end of the string. It makes no distinction between characters after the '#', so it may strip '?' and anything following that. In that case, the "hash" is apparently being treated as "anything from '#', always going to the end of the string," and the query params are being treated as anything _before_ that '#' that is stripped. However, in this method, the hash is matched only as anything from (including) '#' up to the first '?', and not necessarily to the end of the string. So, it seems like there's some inconsistency in what is treated as the "hash" here. What happens to the text that comes after the '#' but also after a '?'. They get stripped in `getQueryString`, but they also don't get matched here, so they seem to be kind of left in a void. Should they be considered part of this hash? If not, how do we handle them?

I don't know the answers to these questions. :)